addons/l10n_ch/tests/test_partner_bank_isr.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2020 Camptocamp SA
```

Comment:
The point of my honoured colleague is to not make a test for every small field, but to test at least a (small) business flow.  If we did this for every field, we could end up with millions of tests, which would render testing unusable.  I will however recheck at ease your pull requests and see what makes sense and what does not.  It does make sense to at least check something about the way the number is composed.  @guewen @gurneyalex  Admittedly, it takes me some time to check your PRs well, but I will.

Reply:
**tl; dr**

To me the pros overweight the cons of having those tests against not enough test.
Tests improves module stability and gives real life data.
IMO simply asking to remove 4 files of tests without alternative is a bad practice. (accross multiple PRs)

----

I understand your point of view. And I must disagree. Here the purpose of tests is to strengthen the code. And this code needs this consolidation to have a clean base. Maybe it's me wanting Swiss quality for some details. But it's not only details. The current state is broken and not having tests is not at all a valid solution.

Those tests don't only test a field constraint. It might not be "a small flow" as you are asking. It is about a part of the process. Which is: "as a user encoding objects in Odoo". It validates the XML view has the correct fields and won't have our users stuck on key features of the localisation.

File `test_partner_bank_isr.py`  checks that you are able to create any kind of bank account we need in Odoo with l10n_ch installed for various purposes.

File `test_vendor_bill_isr.py`  checks a user is able to correctly encode a vendor bill with an ISR reference and ensure we don't mess the whole workflow by entering bad ISR data.

Currently those tests are not run on runbot and **costs you nothing on CI**, but for localisation developers those are helpful and even needed.

Bigger flow tests have the issue of not being tide to a small scope, thus **hard to keep those patches atomic**. And tests should be tided to the fix. It will eases maintainability for any Odoo developers. Without those tests I'm struggling to understand how you want to know for sure the code works.

Please welcome tests with a bit more joy still be critic of them they can probably be improved but don't ask for simple removal. "Overkill" is not the wording I expected when running the l10n_ch current tests are not maintained (as Runbot don't run them I guess it has been a while those were launched) and seeing those tests would greatly help with dataset to give your team how it should work.

Also, it would greatly help if you reminded me the coding guideline of Odoo, I have been contributing since 2011 to this localization and policy might have changed that I'm not aware.
 
And while you make the final decisions some fundamentals are implemented incorrectly for the Switzerland workflow that your team needs to fix one way or another. I'm here to help, but please, in return please help me help you.

-------

Side note: with all my 4 files of tests run on the whole consolidated code of the module (as found here https://github.com/camptocamp/odoo/tree/13.0-new-l10n_ch-consolidated), the test coverage is still not brilliant this module is in a state were it could have some more tests:

```
pytest --disable-warnings -s odoo/src/addons/l10n_ch/tests/ --cov=odoo/src/addons/l10n_ch/models --cov-config=odoo/src/addons/l10n_ch/.coveragerc

----------- coverage: platform linux, python 3.7.3-final-0 -----------
Name                                                        Stmts   Miss  Cover
-------------------------------------------------------------------------------
/odoo/src/addons/l10n_ch/__init__.py                           14      8    43%
/odoo/src/addons/l10n_ch/__manifest__.py                        1      1     0%
/odoo/src/addons/l10n_ch/models/__init__.py                     7      0   100%
/odoo/src/addons/l10n_ch/models/account_bank_statement.py      11      6    45%
/odoo/src/addons/l10n_ch/models/account_invoice.py            151     44    71%
/odoo/src/addons/l10n_ch/models/account_journal.py             28     17    39%
/odoo/src/addons/l10n_ch/models/mail_template.py               32     27    16%
/odoo/src/addons/l10n_ch/models/res_bank.py                    99     23    77%
/odoo/src/addons/l10n_ch/models/res_company.py                 22     12    45%
/odoo/src/addons/l10n_ch/models/res_config_settings.py          9      0   100%
/odoo/src/addons/l10n_ch/tools/__init__.py                      1      0   100%
/odoo/src/addons/l10n_ch/tools/postfinance.py                  26      1    96%
-------------------------------------------------------------------------------
TOTAL                                                         401    139    65%
```

I didn't compare against the 2 current test files in the current state as 1 file is failing and the second one `test_l10n_ch_swissqr.py` is meaning less as it doesn't check QR but checks adress split that is not even used.